data-prepper-core/src/test/java/org/opensearch/dataprepper/pipeline/router/DataFlowComponentRouterTest.java

```diff
@@ -236,6 +247,17 @@ void route_no_Events_when_none_have_matching_routes() {
             verify(componentRecordsConsumer).accept(testComponent, Collections.emptyList());
         }
 
+        @Test
```

Comment:
I think we are missing a test case where you have a default route, but the events are routed to another route. Then we verify it doesn't go to the default.